# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: docshell/base/nsDocShell.cpp
# Commit: 9a8abd87cc79
# Full Hash: 9a8abd87cc7935f29b94248c1a6f8203faa14403
# Author: Henri Sivonen <hsivonen@hsivonen.fi>
# Date: 2021-12-08 21:43:06
# Regressor Bug: 1701828
# File Overlap Count: 10
# Description:
#   Bug 1701828 - meta charset rewrite. r=smaug
#   
#   Implements https://github.com/whatwg/html/issues/6962 . Improves performance
#   when <meta charset> occurs in head but after the first kilobyte and aligns
#   behavior better with WebKit and Blink.
# ==============================================================================

diff --git a/docshell/base/nsDocShell.cpp b/docshell/base/nsDocShell.cpp
--- a/docshell/base/nsDocShell.cpp
+++ b/docshell/base/nsDocShell.cpp
@@ -1591,17 +1591,16 @@ nsDocShell::ForceEncodingDetection() {
             Telemetry::LABELS_ENCODING_OVERRIDE_SITUATION_2::ChannelUtf8);
       } else {
         LOGCHARSETMENU(("ChannelNonUtf8"));
         Telemetry::AccumulateCategorical(
             Telemetry::LABELS_ENCODING_OVERRIDE_SITUATION_2::ChannelNonUtf8);
       }
       break;
     case kCharsetFromXmlDeclaration:
-    case kCharsetFromMetaPrescan:
     case kCharsetFromMetaTag:
       if (isFileURL) {
         LOGCHARSETMENU(("LocalLabeled"));
         Telemetry::AccumulateCategorical(
             Telemetry::LABELS_ENCODING_OVERRIDE_SITUATION_2::LocalLabeled);
       } else if (encoding == UTF_8_ENCODING) {
         LOGCHARSETMENU(("MetaUtf8"));
         Telemetry::AccumulateCategorical(